openqabot/types/incident.py

```diff
@@ -144,3 +148,53 @@ def contains_package(self, requires: List[str]) -> bool:
                 if package.startswith(req) and package != "kernel-livepatch-tools":
                     return True
         return False
+
+    def has_failures(self, token) -> bool:
+        results = requests.get(
+            QEM_DASHBOARD + "/api/jobs/incident/" + str(self.id), headers=token
+        ).json()
+
+        failed_jobs = self.filter_failures(results)
+        if failed_jobs:
+            log.info("Found %s failed jobs for incident %s:", len(failed_jobs), self.id)
+            if log.isEnabledFor(LOG_DEBUG_LEVEL):
+                list(
+                    map(lambda job: self.log_debug_incident(job["job_id"]), failed_jobs)
+                )
+
+        if not results:
+            raise NoResultsError(
+                "Job setting %s not found for incident" % (str(self.id))
+            )
+
+        # we only need to check if there are any failed jobs
+        return any(failed_jobs)
+
+    def log_debug_incident(self, job_id):
+        log.debug(
+            "Job %s is not marked as acceptable for incident %s",
+            job_id,
+            self.id,
+        )
+
+    def filter_failures(self, results):
+        return [
+            res
+            for res in results
+            if res["status"] not in ("passed", "softfailed")
+            and not has_ignored_comment(res["job_id"], self.id)
+        ]
+
+
+@staticmethod
+def has_ignored_comment(job_id: int, inc: int):
+    ret = []
+    ret = requests.get(OPENQA_URL + "/api/v1/jobs/%s/comments" % job_id).json()
+    regex = re.compile(r"\@review\:acceptable_for\:incident_%s\:(.+)" % inc)
+    for comment in ret:
+        if regex.match(comment["text"]):
+            # leave comment for future debugging purposes
+            # log.debug("matched comment incident %s: with comment %s", inc, comment)
```

Comment:
Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
> Don't add disabled code as nobody but you will understand why this is not active.

That only applies if the person looking at the code doesn't read the actual comment stating why it is there

Reply:
Having a commented out log.debug line does not make any sense to me. I am fine to just have the log.debug enabled. If the debug log level is too noisy for anyone then they can set to only log with "info" level

Reply:
What should be the difference between debug and trace level here? I also thing the message is perfectly fine there with debug level. If that message is helpful during debugging (as the comment suggests) it has no meaning to leave that commented out.

Reply:
Also, we shifted some messages between levels in https://github.com/openSUSE/qem-bot/pull/19 and similar and finally ended with limiting the automated pipeline to info level only. So the debug level is indeed intended for any helpful messages for manual debugging.

Reply:
I'd also say this code should be either enabled or removed completely (but wouldn't make a big deal out of it either).

Reply:
In fact, after last changes, this is not necessary anymore and